File: backend-sge/src/main/java/com/basis/sge/dominio/Evento.java
package com.basis.sge.dominio;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "evento")
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class Evento implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @Column(name = "id")
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "sq_evento")
    @SequenceGenerator(name = "sq_evento", sequenceName = "sq_evento", allocationSize = 1)
    private Integer id;

    @Column(name = "local")
    private String local;

    @Column(name = "titulo")
    private String titulo;

    @Column(name = "descricao")
    private String descricao;

    @Column(name = "qtd_vagas")
    private Integer qtdVagas;

    @Column(name = "valor")
    private Double valor;

    @Column(name = "data_inicio")
    private LocalDateTime dataInicio;

    @Column(name = "data_fim")
    private LocalDateTime dataFim;

    @Column(name = "tipo_inscricao")
    private Boolean tipoInscricao;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_tipo_evento")
    private TipoEvento tipoEvento;

    @OneToMany(fetch = FetchType.LAZY, orphanRemoval = true, mappedBy = "evento")
    private List<EventoPergunta> perguntas;


}

File: backend-sge/frontend/src/app/dominios/evento.ts
import { EventoPergunta } from "./evento-pergunta";
import { Pergunta } from "./pergunta";
import { TipoEvento } from "./tipo-evento";
import { TipoSituacao } from './tipo-situacao';

export class Evento {

    id: number;
    titulo: string;
    local: string;
    descricao: string;
    qtdVagas: number;
    valor: number;
    dataInicio: Date;
    dataFim: Date;
    tipoInscricao: boolean;
    tipoEvento: number;
    perguntas: EventoPergunta[] = [];

    constructor() { }
}